base/commands/map/map_test.go

```diff
@@ -0,0 +1,134 @@
+	"github.com/hazelcast/hazelcast-commandline-client/internal/it"
+)
+
+func TestMap(t *testing.T) {
+	testCases := []struct {
+		name string
+		f    func(t *testing.T)
+	}{
+		{name: "EntrySet_NonInteractive", f: entrySet_NonInteractiveTest},
+		{name: "Get_Noninteractive", f: get_NonInteractiveTest},
+		{name: "Set_NonInteractive", f: set_NonInteractiveTest},
+		{name: "Size_Noninteractive", f: size_NoninteractiveTest},
+		{name: "Size_Interactive", f: size_InteractiveTest},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, tc.f)
+	}
+}
+
+func entrySet_NonInteractiveTest(t *testing.T) {
+	mapTester(t, func(tcx it.TestContext, m *hz.Map) {
+		t := tcx.T
+		// no entry
+		tcx.WithReset(func() {
+			check.Must(tcx.CLC().Execute("map", "-n", m.Name(), "entry-set", "--quite"))
+			tcx.AssertStdoutEquals(t, "")
+		})
+		// set an entry
+		tcx.WithReset(func() {
+			check.Must(m.Set(context.Background(), "foo", "bar"))
+			check.Must(tcx.CLC().Execute("map", "-n", m.Name(), "entry-set", "--quite"))
+			tcx.AssertStdoutContains(t, "foo\tbar\n")
+		})
+		// show type
+		tcx.WithReset(func() {
+			check.Must(tcx.CLC().Execute("map", "-n", m.Name(), "entry-set", "--show-type", "--quite"))
+			tcx.AssertStdoutContains(t, "foo\tSTRING\tbar\tSTRING\n")
+		})
+	})
+}
+
+func get_NonInteractiveTest(t *testing.T) {
+	mapTester(t, func(tcx it.TestContext, m *hz.Map) {
+		t := tcx.T
+		// no entry
+		tcx.WithReset(func() {
+			check.Must(tcx.CLC().Execute("map", "-n", m.Name(), "get", "foo", "--quite"))
+			tcx.AssertStdoutEquals(t, "-\n")
+		})
+		// set an entry
+		tcx.WithReset(func() {
+			check.Must(m.Set(context.Background(), "foo", "bar"))
+			check.Must(tcx.CLC().Execute("map", "-n", m.Name(), "get", "foo", "--quite"))
+			tcx.AssertStdoutContains(t, "bar\n")
+		})
+	})
+}
+
+func set_NonInteractiveTest(t *testing.T) {
+	mapTester(t, func(tcx it.TestContext, m *hz.Map) {
+		t := tcx.T
+		tcx.WithReset(func() {
+			check.Must(tcx.CLC().Execute("map", "-n", m.Name(), "set", "foo", "bar", "--quite"))
+			tcx.AssertStdoutEquals(t, "")
+			v := check.MustValue(m.Get(context.Background(), "foo"))
+			require.Equal(t, "bar", v)
+		})
+	})
+}
+
+func size_NoninteractiveTest(t *testing.T) {
+	mapTester(t, func(tcx it.TestContext, m *hz.Map) {
+		t := tcx.T
+		// no entry
+		tcx.WithReset(func() {
+			check.Must(tcx.CLC().Execute("map", "-n", m.Name(), "size", "--quite"))
+			tcx.AssertStdoutEquals(t, "0\n")
+		})
+		// set an entry
+		tcx.WithReset(func() {
+			tcx.AssertStdoutEquals(t, "")
+			check.Must(m.Set(context.Background(), "foo", "bar"))
+			check.Must(tcx.CLC().Execute("map", "-n", m.Name(), "size", "--quite"))
+			tcx.AssertStdoutEquals(t, "1\n")
+		})
+	})
+}
+
+func size_InteractiveTest(t *testing.T) {
+	mapTester(t, func(tcx it.TestContext, m *hz.Map) {
+		t := tcx.T
+		ctx := context.Background()
+		go func(t *testing.T) {
+			check.Must(tcx.CLC().Execute())
+		}(t)
+		tcx.WithReset(func() {
```

Comment:
IMHO we can embed this logic to a function (maybe inside TestContext) so we don't have to write this in every interactive test